### PR TITLE
Fix issue of connecting to influxdb without username/password on okhttp 4.x.x

### PR DESCRIPTION
**Pull request by vinh0604**
This pull request is to fix the bug mentioned in #643 by only add `BasicAuthInterceptor` to `OkHttpClient.Builder` when both username and password are not null.

### CODE REVIEW

**Comment by majst01** [comment]
Hi @vinh0604 

thanks for this fix, i was experiencing the same when i tried to upgrade okhttp as part of my regular dependency updates. But i was to lazy to look into the root cause. So welcome !

actually all test are falling with:
```
[ERROR] /usr/src/mymaven/src/main/java/org/influxdb/impl/InfluxDBImpl.java:47: Using the '.*' form of import should be avoided - java.util.*. [AvoidStarImport]
```
this must be addressed first. I also have some small nits on the MR.

File: src/main/java/org/influxdb/impl/BatchProcessor.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ public Builder threadFactory(final ThreadFactory threadFactory) {
      *            is mandatory.
      */
     public Builder(final InfluxDB influxDB) {
-      this.influxDB = (InfluxDBImpl) influxDB;
+      this.influxDB = (InfluxDB) influxDB;
```
Comment (majst01) [inline src/main/java/org/influxdb/impl/BatchProcessor.java]: I think this cast can now be removed

File: src/main/java/org/influxdb/impl/InfluxDBImpl.java [hunk 2]
```diff
@@ -44,10 +44,7 @@
 import java.net.URISyntaxException;
 import java.net.UnknownHostException;
 import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
```
Comment (majst01) [inline src/main/java/org/influxdb/impl/InfluxDBImpl.java]: this causes the build to fail, no wildcard imports please.

**Comment by vinh0604** [comment]
Sorry, I didn't run stylecheck before pushing. Those issues were resolved now. 
Please check.

**Comment by majst01** [comment]
Now all checks have passed, great. May i asked you to add a additional unit test case to ensure that the extra if statement is covered. I really appreciate your contribution !

**Comment by vinh0604** [comment]
> Now all checks have passed, great. May i asked you to add a additional unit test case to ensure that the extra if statement is covered. I really appreciate your contribution !

Thanks, I added one more test case in `InfluxDBFactoryTest` as well as removing the if condition in the internal construct of `InfluxDBImpl` that is only used in test.

**Comment by majst01** [comment]
Thanks for contributing.

**Pull request merged**